### PR TITLE
REVERTED: Conditionally select a field to return in ResponseParser for InfluxDB

### PR DESCRIPTION
**Pull request by lexmag**
This is a possible fix for #7646 (the field selection could be added later if necessary).
I'm not sure why `value[0]` was ever used before, but for SELECT queries InfluxDB always returns "time" first.
Also it is not possible to request just time—non-time field is required.

If, for some reason, returning time is needed I think we could do the following change:
```javascript
var isReturningTime = query.toLowerCase().indexOf('time as') >= 0;
var position = if (isReturningTime) { 0 } else { 1 };
addUnique(res, value[position]);		
```
What are your thought on it?

_CC @melinlab_

### CODE REVIEW

**Comment by torkelo** [comment]
this looks good, can you add unit test for this?

**Comment by torkelo** [comment]
https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/influxdb/specs/response_parser.jest.ts 

run test with `npm run jest`

**Comment by lexmag** [comment]
Sure, on it.

**Comment by bergquist** [comment]
If I remember correctly this was a breaking change between 0.10 and 0.11 (hence the variable name).

**Comment by lexmag** [comment]
@bergquist according to tests it looks like there is a breaking change between 0.10 and 0.11. Wondering what is the current minimum InfluxDB version requirement?

**Comment by bergquist** [comment]
@lexmag yeap. Since its so easy to keep supporting 0.10 I think we should. A comment in the code would be great. Should make it easier for others to understand in the future.

**Comment by lexmag** [comment]
Updated with the requested changes.

File: public/app/plugins/datasource/influxdb/specs/response_parser.jest.ts [hunk 1]
```diff
@@ -85,6 +85,32 @@ describe('influxdb response parser', () => {
...
+          ],
+        },
+      ],
+    };
+
+    var result = parser.parse(query, response);
+
+    it('should return second column', () => {
+      expect(_.size(result)).toBe(3);
+      expect(result[0].text).toBe(10.0);
```
Comment (torkelo) [inline public/app/plugins/datasource/influxdb/specs/response_parser.jest.ts]: should text not be of string type?

Comment (lexmag) [reply public/app/plugins/datasource/influxdb/specs/response_parser.jest.ts]: As I see `result[0]` contains `{ text: 10 }`, thus we need a `.text` call.

Comment (torkelo) [reply public/app/plugins/datasource/influxdb/specs/response_parser.jest.ts]: text property should always be a string

Comment (lexmag) [reply public/app/plugins/datasource/influxdb/specs/response_parser.jest.ts]: @torkelo I've fixed that in the next commit. The fix seems non-strictly related to the PR—I could send it separately in another PR if you would like.

**Comment by lexmag** [comment]
I wonder, is it possible to get it into the upcoming release?

File: public/app/plugins/datasource/influxdb/response_parser.ts [hunk 2]
```diff
@@ -11,25 +11,30 @@ export default class ResponseParser {
...
+          // where the first element is somewhat a metadata value:
+          // the tag name for SHOW TAG VALUES queries,
+          // the time field for SELECT queries, etc.
+          // The second shape is an one-element array,
+          // that is containing an immediate value.
+          // For example, SHOW FIELD KEYS queries return such shape.
+          // Note, pre-0.11 versions return
+          // the second shape for SHOW TAG VALUES queries
+          // (while the newer versions—first).
+          addUnique(res, value[1] || value[0]);
```
Comment (torkelo) [inline public/app/plugins/datasource/influxdb/response_parser.ts]: if value value[1] is "0" or 0 then it will be ignored.

Comment (lexmag) [reply public/app/plugins/datasource/influxdb/response_parser.ts]: @torkelo thanks, it is fixed now.

**Comment by lexmag** [comment]
Any more things I need to fix to get it merged? :bowtie:

**Comment by lexmag** [comment]
Ping. :bowtie:

**Comment by bergquist** [comment]
Thank you for your patience. All add this to my review queue. Its very long atm so please be patient >.<

**Comment by daniellee** [comment]
@lexmag Going to have revert this due to this error: #12009

image

The changed conditional means that this [line](https://github.com/grafana/grafana/blob/92e5968f91472abdba17405092e092a532de5eb4/public/app/plugins/datasource/influxdb/response_parser.ts#L30) sets the wrong value.

**Pull request merged**